sssd_test_framework/utils/adcli.py

```diff
@@ -0,0 +1,168 @@
+    for clean up.
+    """
+
+    def info(self, domain: str, *, args: list[str] | None = None) -> ProcessResult:
+        """
+        Discover AD domain.
+
+        :param domain: domain.
+        :type domain: str,
+        :param args: additional arguments, defaults to None
+        :type args: list[str] | None, optional
+        """
+        if args is None:
+            args = []
+
+        return self.host.conn.exec(["adcli", "info", *args, "--verbose", f"{domain}"])
+
+    def testjoin(self, *, args: list[str] | None = None) -> ProcessResult:
+        """
+        Validate join.
+
+        :param args: additional arguments, defaults to None
+        :type args: list[str] | None, optional
+        """
+        if args is None:
+            args = []
+
+        return self.host.conn.exec(["adcli", *args, "--verbose", "testjoin"])
+
+    def delete_computer(
+        self,
+        host: str,
+        *,
+        args: list[str] | None = None,
+        password: str | None = None,
+    ) -> ProcessResult:
+        """
+        Delete computer account.
+
+        :param host: client hostname
+        :type host: str | None,
+        :param args: additional arguments, defaults to None.
+        :type args: list[str] | None, optional
+        :param password: password, defaults to None.
+        :type password: str | None, optional
+        """
+        if args is None:
+            args = []
+
+        return self.host.conn.exec(["adcli", "delete-computer", *args, host], input=password)
+
+    def join(
+        self,
+        domain: str,
+        *,
+        args: list[str] | None = None,
+        password: str | None = None,
+        user: str | None = None,
+        krb: bool | None = False,
+        prompt_password: bool | None = False,
+        stdin: bool | None = False,
+    ) -> ProcessResult:
+        """
+        Create a computer account.
+
+        :param domain: domain.
+        :type domain: str,
+        :param args: additional arguments, defaults to None
+        :type args: list[str] | None, optional
+        :param password: password, defaults to None
+        :type password: str | None, optional
+        :param user: Authenticating User, defaults to None
+        :type user: str | None, optional
+        :param krb: Use kerberos credentials.
+        :type krb: bool | False, optional
+        """
+        if args is None:
+            args = []
+
+        if krb:
+            self.host.conn.exec(["kinit", user], input=password)
+            command = self.host.conn.exec(["adcli", "join", "--verbose", "--login-ccache", *args, domain])
+        elif prompt_password:
+            command = self.host.conn.exec(["adcli", "join", "-W", "--verbose", *args, domain], input=password)
+        elif stdin:
+            command = self.host.conn.exec(
+                ["echo", password, "|", "adcli", "join", "--stdin-password", "--verbose", *args, domain]
+            )
+
+        return command
+
+    def create_user(
```

Comment:
We might want to have also passwd-user so we can set password and try to login as the user.